sdk/core/Azure.Core/src/Shared/ClientDiagnostics.cs

```diff
@@ -34,7 +41,28 @@ public DiagnosticScope CreateScope(string name)
             {
                 return default;
             }
-            return new DiagnosticScope(name, _source);
+            var scope = new DiagnosticScope(name, _source);
+
+            if (_resourceProviderNamespace != null)
+            {
+                scope.AddAttribute("az.namespace", _resourceProviderNamespace);
+            }
+            return scope;
+        }
+
+        internal static string? GetResourceProviderNamespace(Assembly assembly)
+        {
+            foreach (var customAttribute in assembly.GetCustomAttributes(true))
+            {
+                // Weak bind internal shared type
+                var attributeType = customAttribute.GetType();
+                if (attributeType.Name == "AzureResourceProviderNamespaceAttribute")
+                {
+                    return attributeType.GetProperty("ResourceProviderNamespace")?.GetValue(customAttribute) as string;
```

Comment:
I'd use `nameof` in the unlikely event it gets refactored.

Reply:
I'm trying to avoid dependency on `AzureResourceProviderNamespaceAttribute` type here because it's a shared-source type.